flang/test/Driver/main_dupes.f90

```diff
@@ -0,0 +1,15 @@
+! UNSUPPORTED: system-windows
+
+! RUN: %flang -x ir -o %t.c-object -c %S/Inputs/main_dupes.ll
+! RUN: %flang -o %t -c %s
+! RUN: not %flang -o %t.exe %t %t.c-object 2>&1
```

Comment:
[nit] You could add a "valid"/"working" compilation line to contrast it with the broken one.

Reply:
I'd actually prefer to have a separate test for this, as a working test is any other test that actually produces an executable.

If you insist :-), then I'd even go and change the test from testing for a duplicate of main to a general link time test that tests both a successful link and the failing one that I'm after.

Reply:
That was just a nit - I am happy if you keep things as is 👍🏻 !